sdk/azidentity/client_certificate_credential_test.go

```diff
@@ -85,3 +108,79 @@ func TestClientCertificateCredential_GetTokenInvalidCredentials(t *testing.T) {
 		t.Fatalf("Expected an error but did not receive one")
 	}
 }
+
+func TestClientCertificateCredential_CreateAuthRequestFail(t *testing.T) {
+	srv, close := mock.NewServer()
+	defer close()
+	srv.SetResponse(mock.WithStatusCode(http.StatusUnauthorized))
+	srvURL := srv.URL()
+	srvURL.Host = "ht @"
+	cred, err := NewClientCertificateCredential(tenantID, clientID, certificatePath, &TokenCredentialOptions{HTTPClient: srv, AuthorityHost: &srvURL})
+	if err != nil {
+		t.Fatalf("Unable to create a ClientCertificateCredential")
+	}
+	_, err = cred.GetToken(context.Background(), azcore.TokenRequestOptions{Scopes: []string{scope}})
+	if err == nil {
+		t.Fatalf("Expected an error but did not receive one")
+	}
+}
+
+func TestClientCertificateCredential_GetTokenCheckPrivateKeyBlocks(t *testing.T) {
+	srv, close := mock.NewServer()
+	defer close()
+	srv.AppendResponse(mock.WithBody([]byte(`{"access_token": "new_token", "expires_in": 3600}`)))
+	srvURL := srv.URL()
+	cred, err := NewClientCertificateCredential(tenantID, clientID, "certificate_formatB.pem", &TokenCredentialOptions{HTTPClient: srv, AuthorityHost: &srvURL})
```

Comment:
I think the path to the pem files will need to be updated too.

Reply:
Done. I did add the file to my last commit.